index.js

```diff
@@ -312,6 +312,7 @@ function fromArrayBuffer (array, byteOffset, length) {
 
 function fromObject (obj) {
   if (Buffer.isBuffer(obj)) {
+    // Note: Probably not necessary anymore.
```

Comment:
What isn't necessary about this now?

Reply:
Because `ArrayBuffer.isView` returns true for `Uint8Array` subclasses like `Buffer`. So this case will actually be handled earlier, in `Buffer.from`:

``` js
  if (ArrayBuffer.isView(value)) {
    return fromArrayView(value)
  }
```

I left the `Buffer.isBuffer(obj)` check in there because I was concerned that maybe non-v8 javascript engines' `ArrayBuffer.isView` does not return `true` for `Uint8Array` subclasses (from the same context or another). 

Because `ArrayByffer.isView` is so magic in v8, I actually wanted to test it in FF and Safari to make sure they had the same behavior, but I didn't get around to it, so leaving the check in is just me being cautious.

Reply:
You should add comments like ^ into your code :blue_heart:

Reply:
Yeah, it's something I need to work on.